### PR TITLE
PNI 'callout-box' style signup forms are now rendered using the new React components

### PR DESCRIPTION
**Pull request by mmmavis**
Related PRs/issues: #11578 

## Changes that affect all existing "default style" form fields on the site:
👋🏻 Hello design reviewer @nancyt1 
I've made some [form field styling adjustments](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/11636/files#diff-15eb35d3f98a5be49e8be397d371a0f3b4c5b6780ba281f14fce904b5e88dd0f) based on [design system on Figma](https://www.figma.com/file/sOxNGYMzwZT0B9Fal2D4WT/%5BWIP%5D-Component-Library?node-id=1232%3A11585&mode=dev) so the the button height and input field height look more cohesive when they are side by side. (Changes involved: updated padding, increased input font size to 18px and line height to 1.5, updated placeholder text colour to gray 40). Note that these are just quick fixes and we'll still need to clean up our form styling config holistically in the codebase in the future

## Updated the following "callout-box" style forms to use the new React components.

image

### PNI Homepage (in product grid)
Review app: https://foundation-s-11578-news-ayqkwr.herokuapp.com/en/privacynotincluded/  
Prod: https://foundation.mozilla.org/en/privacynotincluded/  

### PNI product page
Review app: https://foundation-s-11578-news-ayqkwr.herokuapp.com/en/privacynotincluded/able-treat-pattern/
Prod: https://foundation.mozilla.org/en/privacynotincluded/euki/  

### PNI article index page
Review app: https://foundation-s-11578-news-ayqkwr.herokuapp.com/en/privacynotincluded/articles/ 
Prod: https://foundation.mozilla.org/en/privacynotincluded/articles/ 

### PNI article page
Review app: https://foundation-s-11578-news-ayqkwr.herokuapp.com/en/privacynotincluded/articles/information-these-break-represent-me-still-though-see/
Prod: https://foundation.mozilla.org/en/privacynotincluded/articles/our-2023-nice-list-we-see-you-good-at-privacy-connected-products/

### CODE REVIEW

File: tailwind-plugins/form-control.js [hunk 1]
```diff
@@ -18,12 +18,12 @@ function formControl(theme) {
   return {
     display: "block",
     width: "100%",
-    height: "calc(1.5em + .75rem + 2px)",
-    padding: ".375rem .75rem",
+    height: "calc( 2.5rem + 2px)", // 42px if 1rem = 16px
```
Comment (mmmavis) [inline tailwind-plugins/form-control.js:21]: Based on values in our design system: https://www.figma.com/file/sOxNGYMzwZT0B9Fal2D4WT/%5BWIP%5D-Component-Library?node-id=1232%3A11585&mode=dev

File: source/js/components/newsletter-signup/organisms/default-layout-signup.jsx [hunk 2]
```diff
@@ -233,4 +236,11 @@ DefaultSignupForm.propTypes = {
   hideForm: PropTypes.bool,
 };
 
+/**
+ * This renders a signup form which includes the following fields
```
Comment (danielfmiranda) [inline source/js/components/newsletter-signup/organisms/default-layout-signup.jsx:240]: Thanks for leaving comments! :raised_hands:

**Review by danielfmiranda — APPROVED** [review]
Looks great to me! Thanks @mmmavis 👍 approved

**Review by nancyt1 — APPROVED** [review]
Looks good to me!

**Pull request merged**